tools/test-npm.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+set -e
+
+# always change the working directory to the project's root directory
+cd $(dirname $0)/..
+
+# pass NODE_EXE from something like Makefile
+# it should point to either ./node or ./node.exe, depending on the platform
+if ! [ -n "$NODE_EXE" ]; then
+  echo "No node executable provided. Bailing." >&2
+  exit 0
+fi
+
+rm -rf test-npm
+mkdir test-npm
+
+# make a copy of deps/npm to run the tests on
+cp -r deps/npm/ test-npm/
+
+cd test-npm
+
+# do a rm first just in case deps/npm contained these
+rm -rf npm-cache npm-tmp npm-prefix
+mkdir npm-cache npm-tmp npm-prefix
+
+# these commands all require the same env variables, so run them as one
+npm_config_cache="npm-cache" \
+npm_config_prefix="npm-prefix" \
+npm_config_tmp="npm-tmp" \
+../$NODE_EXE cli.js install --ignore-scripts && \
+../$NODE_EXE cli.js run-script test-all && \
+../$NODE_EXE cli.js prune --prod
```

Comment:
As a script, if I'm not mistaken you can just `export npm_config_cache="bloo-blee-bloo"` and it won't affect the outer environment. Then you can un-chain the cli runs, since `set -e` is enabled.